docs/usage/applications.rst

```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
+    or a :class:`list` of :class:`tuples <tuple>` containing key/value pairs.
 
-.. note::
-
-    :class:`State <.datastructures.state.State>` can be initialized with a dictionary, an instance of
-    :class:`ImmutableState <.datastructures.state.ImmutableState>` or :class:`State <.datastructures.state.State>`,
-    or a list of tuples containing key/value pairs.
-
-.. attention::
-
-    You may instruct :class:`State <.datastructures.state.State>` to deep copy initialized data to prevent mutation from
-    outside the application context. To do this, pass ``deep_copy=True`` to the :class:`State <.datastructures.state.State>`
-    constructor.
+.. attention:: You may instruct :class:`~.datastructures.state.State` to deep copy initialized data to prevent mutation
+    from outside the application context. To do this, st :paramref:`~.datastructures.state.State.deep_copy` to
+    ``True`` in the :class:`~.datastructures.state.State` constructor.
 
 Injecting Application State into Route Handlers and Dependencies
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 As seen in the above example, Litestar offers an easy way to inject state into route handlers and dependencies - simply
-by specifying ``state`` as a kwarg to the handler function. I.e., you can simply do this in handler function or dependency
-to access the application state:
+by specifying ``state`` as a kwarg to the handler function. In other words, you can simply do this in handler function
+or dependency to access the application state:
 
 .. code-block:: python
+    :caption: Accessing application :class:`~.datastructures.state.State` in a handler function
 
-   from litestar import get
-   from litestar.datastructures import State
+    from litestar import get
+    from litestar.datastructures import State
 
 
-   @get("/")
-   def handler(state: State) -> None: ...
+    @get("/")
+    def handler(state: State) -> None: ...
 
 When using this pattern you can specify the class to use for the state object. This type is not merely for type
-checkers, rather Litestar will instantiate a new state instance based on the type you set there. This allows users to
-use custom classes for State, e.g.:
+checkers, rather Litestar will instantiate a new ``state`` instance based on the type you set there.
+This allows users to use custom classes for :class:`~.datastructures.state.State`, e.g.:
 
-While this is very powerful, it might encourage users to follow anti-patterns: it's important to emphasize that using
-state can lead to code that's hard to reason about and bugs that are difficult to understand, due to changes in
+While this is very powerful, it might encourage users to follow anti-patterns: it is important to emphasize that using
+state can lead to code that is hard to reason about and bugs that are difficult to understand, due to changes in
 different ASGI contexts. As such, this pattern should be used only when it is the best choice and in a limited fashion.
```

Comment:
It feels weird to me that in an introductory usage document, we effectively say, "Hey, here's a thing you can do with `State`, but its weird and bug-prone, so don't do it."

Makes me feel that we should have a separate section in usage for `State` that explores these concepts outside of the app use docs. WDYT?

Reply:
What's a good spot you think? a topic? or a section in this further down maybe under `Advanced Stuff`?

Reply:
Opened #3346 for this